eth/db/batch.py

```diff
@@ -74,3 +80,88 @@ def __delitem__(self, key: bytes) -> None:
 
     def diff(self) -> DBDiff:
         return self._track_diff.diff()
+
+
+class BatchableDB(BaseBatchableDB):
+    """
+    This is nearly the same as BatchDB, but it immediately writes out changes if they are
+    not in a batch_write() context.
+    """
+    logger = logging.getLogger("eth.db.BatchableDB")
+
+    wrapped_db = None  # type: BaseDB
+    _track_diff = None  # type: DBDiffTracker
+
+    def __init__(self, wrapped_db: BaseDB = None) -> None:
+        if wrapped_db is None:
+            self.wrapped_db = MemoryDB()
+        else:
+            self.wrapped_db = wrapped_db
+        self._track_diff = DBDiffTracker()
+        self._is_batching = False
```

Comment:
I think this will be cleaner if you use a `threading.Lock` since they support context manager APIs which should cleanup the internals of `write_batch`.